src/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -535,7 +535,7 @@ public static void UtcDateTime()
         [Fact]
         public static void UtcNow()
         {
-            Assert.NotSame(DateTimeOffset.UtcNow, DateTimeOffset.UtcNow);
+            Assert.NotNull(DateTimeOffset.UtcNow);
```

Comment:
DateTimeOffset is a struct.  NotNull takes an object.  This will box the struct and then it'll never be null.

How about verifying that the value changes in a positive direction, e.g.
```C#
DateTimeOffset start = DateTimeOffset.UtcNow;
Assert.True(
    SpinWait.SpinUntil(() => DateTimeOffset.UtcNow > start, TimeSpan.FromSeconds(2),
    "Expected UtcNow to change");
```